api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -3238,4 +3239,38 @@ public void mergePatients_shouldMoveAllObsWithSameHierarchy() throws Exception {
 		assertEquals(2, encounterService.getEncounter(57).getObsAtTopLevel(true).size());
 	}
 
+	/**
+	 * @see PatientServiceImpl#getDuplicatePatientsByAttributes(List) 
+	 */
+	@Test(expected = APIException.class)
+	public void getDuplicatePatientsByAttributes_shouldThrowAnAPIExceptionIfAttributesIsNull() {
+		List<String> attributes = null;
+		patientService.getDuplicatePatientsByAttributes(attributes);
+	}
+
+	/**
+	 * @see PatientServiceImpl#getDuplicatePatientsByAttributes(List)
+	 */
+	@Test(expected = APIException.class)
+	public void getDuplicatePatientsByAttributes_shouldThrowAnAPIExceptionIfAttributesIsEmpty() {
+		List<String> attributes = new ArrayList<>();
+		patientService.getDuplicatePatientsByAttributes(attributes);
+	}
+
+	/**
+	 * @see PatientServiceImpl#getDuplicatePatientsByAttributes(List)
+	 * attributes on a Person or Patient object. similar to: [gender, givenName,
+	 *            middleName, familyName]
+	 */
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatients() {
+		Patient patient = Context.getPatientService().getPatient(2);
+		List<String> attributes = new ArrayList<>();
+		attributes.add(patient.getGender());
+		attributes.add(patient.getGivenName());
+		attributes.add(patient.getMiddleName());
+		attributes.add(patient.getFamilyName());
+		List<Patient> receivedPatientList = patientService.getDuplicatePatientsByAttributes(attributes);
```

Comment:
Can you please test with actual attributes in the database that return some matches?